packages/runtime/container-runtime/src/test/blobManager.spec.ts

```diff
@@ -353,6 +357,75 @@ describe("BlobManager", () => {
 		assert.strictEqual(summaryData.redirectTable.size, 3);
 	});
 
+	it("abort before upload", async () => {
+		await runtime.attach();
+		await runtime.connect();
+		const ac = new AbortController();
+		ac.abort("abort test");
+		let handleP;
+		try {
+			const blob = IsoBuffer.from("blob", "utf8");
+			handleP = runtime.createBlob(blob, ac.signal);
+			await runtime.processAll();
+			assert.fail("Should not succeed");
+		} catch (error: any) {
+			assert.strictEqual(error.message, "aborted before uploading");
+		}
+		assert(handleP);
+		await assert.rejects(handleP);
+	});
+
+	it("abort while upload", async () => {
```

Comment:
i would expect this test to be much more detailed. this class defines the mock runtime used. i would expect the test to observe, and delay the call to the mock storage on the mock runtime, and affirmative ensure that abort is call during the createBlob call. this may entail building new mocks.

Reply:
Reading MockRuntime.getStorage.createBlob, it's my understanding that we're delaying the call to the mock storage via processBlobsP promise. We could ensure abort is being called during the createBlob call by checking if there are unprocessed blobs on abort call. I've been trying to extend current MockRuntime but feeling like I would just reuse all functionality or changing just how we wait while creating the blob using a delay instead of a promise and not changing real test scenario.
The missing part would be failed upload in which case I could just create another function failCreateBlob where I can directly reject the mock storage promise.